lib/models/organization.ts

```diff
@@ -0,0 +1,216 @@
+	 *
+	 * @description This method creates a new organization with current user as an administrator.
+	 *
+	 * @param {Object} options - Organization parameters to use.
+	 * @param {String} options.name - Required: the name of the organization that will be created.
+	 * @param {String} [options.handle] - The handle of the organization that will be created.
+	 *
+	 * @fulfil {String} - Organization
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.organization.create({ name:'MyOrganization' }).then(function(organization) {
+	 * 	console.log(organization);
+	 * });
+	 *
+	 * @example
+	 * balena.models.organization.create({ name:'MyOrganization' }, function(error, organization) {
+	 * 	if (error) throw error;
+	 * 	console.log(organization);
+	 * });
+	 */
+	const create = function (
+		organization: BalenaSdk.PineSubmitBody<BalenaSdk.Organization>,
+	): Promise<BalenaSdk.Organization> {
+		return pine.post<BalenaSdk.Organization>({
+			resource: 'organization',
+			body: organization,
+		});
+	};
+
+	/**
+	 * @summary Get all Organizations
+	 * @name getAll
+	 * @public
+	 * @function
+	 * @memberof balena.models.organization
+	 *
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object[]} - organizations
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.organization.getAll().then(function(organizations) {
+	 * 	console.log(organizations);
+	 * });
+	 *
+	 * @example
+	 * balena.models.organization.getAll(function(error, organizations) {
+	 * 	if (error) throw error;
+	 * 	console.log(organizations);
+	 * });
+	 */
+	const getAll = function (
+		options: BalenaSdk.PineOptions<BalenaSdk.Organization> = {},
+	): Promise<BalenaSdk.Organization[]> {
+		return pine.get<BalenaSdk.Organization>({
+			resource: 'organization',
+			options: mergePineOptions(
+				{
+					$orderby: 'name asc',
+				},
+				options,
+			),
+		});
+	};
+
+	/**
+	 * @summary Get a single organization
+	 * @name get
+	 * @public
+	 * @function
+	 * @memberof balena.models.organization
+	 *
+	 * @param {String|Number} handleOrId - organization handle (string) or id (number).
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object} - organization
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.organization.get('myorganization').then(function(organization) {
+	 * 	console.log(organization);
+	 * });
+	 *
+	 * @example
+	 * balena.models.organization.get(123).then(function(organization) {
+	 * 	console.log(organization);
+	 * });
+	 *
+	 * @example
+	 * balena.models.organization.get('myorganization', function(error, organization) {
+	 * 	if (error) throw error;
+	 * 	console.log(organization);
+	 * });
+	 */
+	const get = function (
+		handleOrId: string | number,
+		options: BalenaSdk.PineOptions<BalenaSdk.Organization> = {},
+	) {
+		return Promise.try(() => {
+			if (handleOrId == null) {
+				throw new errors.BalenaOrganizationNotFound(handleOrId);
```

Comment:
Is there a more appropriate error to throw here? `handleOrId` being `null` is more like an "invalid argument" or "bad request" kind of error.

Reply:
Indeed, BalenaInvalidParameterError is the way to go :+1: